### PR TITLE
Fix reading too many columns for s3 and url storages

### LINKED ISSUE #34163: Columns are not skipped while reading Parquet file from s3
**Describe the situation**
```
ip-172-31-0-100.eu-central-1.compute.internal :) SELECT count() FROM s3('https://s3.eu-central-1.amazonaws.com/clickhouse-public-datasets/hits.parquet', '...', '...', Parquet, 'auto')

SELECT count()
FROM s3('https://s3.eu-central-1.amazonaws.com/clickhouse-public-datasets/hits.parquet', '...', '...', Parquet, 'auto')

Query id: d10b6c7f-76d5-437d-a478-3b8291fe187c

┌─count()─┐
│ 8873898 │
└─────────┘

1 rows in set. Elapsed: 33.648 sec. Processed 8.87 million rows, 9.65 GB (263.73 thousand rows/s., 286.80 MB/s.)

ip-172-31-0-100.eu-central-1.compute.internal :) SELECT uniq(CounterID) FROM s3('https://s3.eu-central-1.amazonaws.com/clickhouse-public-datasets/hits.parquet', '...', '...', Parquet, 'auto')

SELECT uniq(CounterID)
FROM s3('https://s3.eu-central-1.amazonaws.com/clickhouse-public-datasets/hits.parquet', '...', '...', Parquet, 'auto')

Query id: bb935e10-3d93-4a1d-8326-24eb9587cc2c

┌─uniq(CounterID)─┐
│          109215 │
└─────────────────┘

1 rows in set. Elapsed: 29.296 sec. Processed 8.87 million rows, 9.65 GB (302.90 thousand rows/s., 329.40 MB/s.)
```

Although it works perfectly for `file` table function:

```
ubuntu@ip-172-31-0-100:~$ clickhouse-local 
ClickHouse local version 22.2.1.3625.

ip-172-31-0-100.eu-central-1.compute.internal :) SELECT uniq(CounterID) FROM file('hits.parquet')

SELECT uniq(CounterID)
FROM file('hits.parquet')

Query id: fdd6887e-bd2d-48d9-a41f-61d879cd5d63

┌─uniq(CounterID)─┐
│          109215 │
└─────────────────┘

1 rows in set. Elapsed: 0.688 sec. Processed 7.12 million rows, 56.98 MB (10.35 million rows/s., 82.78 MB/s.)

ip-172-31-0-100.eu-central-1.compute.internal :) SELECT count() FROM file('hits.parquet')

SELECT count()
FROM file('hits.parquet')

Query id: 83d69839-d3bf-48ce-8c76-641211ccb4f2

┌─count()─┐
│ 8873898 │
└─────────┘

1 rows in set. Elapsed: 0.562 sec. Processed 6.38 million rows, 6.38 MB (11.36 million rows/s., 11.36 MB/s.)
```

### PR DESCRIPTION
**Pull request by kssenii**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix reading only columns which user asked for. Closes https://github.com/ClickHouse/ClickHouse/issues/34163.

### CODE REVIEW

File: src/Storages/StorageS3.cpp [hunk 1]
```diff
@@ -615,6 +615,11 @@ std::shared_ptr<StorageS3Source::IteratorWrapper> StorageS3::createFileIterator(
     }
 }
 
+bool StorageS3::isColumnOriented() const
+{
+    return format_name != "Distributed" && FormatFactory::instance().checkIfFormatIsColumnOriented(format_name);
```
Comment (KochetovNicolai) [inline src/Storages/StorageS3.cpp:620]: I already don't remember what is `Distributed` format. Maybe we should prohibit it from using with anything expect `StorageFile`.

**Review by KochetovNicolai — APPROVED** [review]
Is it possible to add a test?

**Comment by kssenii** [comment]
@Mergifyio update

**Comment by kssenii** [comment]
@Mergifyio update

**Pull request merged**